src/components/OptionsDialog/index.js

```diff
@@ -0,0 +1,109 @@
+import $ from 'jquery'
+import React from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+
+import ReactDOM from 'react-dom'
+
+import FieldHelp from '../FieldHelp/index'
+
+import {
+  getSSHKey,
+  saveSSHKey,
+} from './actions'
+
+class OptionsDialog extends React.Component {
+
+  constructor (props) {
+    super(props)
+    this.state = {
+      sshKey: props.optionsDialog.get('sshKey') || '',
+    }
+    this.onSSHKeyChange = this.onSSHKeyChange.bind(this)
+    this.onSaveClick = this.onSaveClick.bind(this)
+  }
+
+  componentDidMount () {
+    const dom = ReactDOM.findDOMNode(this)
+    $(dom).on('shown.bs.modal', function (e) {
+      if (this.props.userId) {
+        this.props.onOpen()
+      }
+    }.bind(this))
+  }
+
+  componentWillReceiveProps (nextProps) {
+    this.setState({ 'sshKey': nextProps.optionsDialog.get('sshKey') })
+  }
+
+  onSSHKeyChange (event) {
+    this.setState({ sshKey: event.target.value })
+  }
+
+  onSaveClick () {
+    this.props.onSave({ key: this.state.sshKey, sshId: this.props.optionsDialog.get('sshId') })
+  }
+
+  render () {
+    let { oVirtApiVersion } = this.props
+    let content = (
+      <form>
+        <div className='form-group'>
+          <FieldHelp content='Your public SSH key.' text='SSH Key' />
+          <textarea type='text' className='form-control' rows='7' onChange={this.onSSHKeyChange} value={this.state.sshKey || ''} />
+        </div>
+      </form>
+    )
+
+    if (!this.props.userId) {
+      let apiVersion = 'unknown'
+      if (oVirtApiVersion && oVirtApiVersion.get('major')) {
+        apiVersion = `${oVirtApiVersion.get('major')}.${oVirtApiVersion.get('minor')}`
+      }
+
+      content = (
+        <div>
+          <p>You can't manage SSH keys, because current oVirt version is to old (oVirt <strong>{apiVersion}</strong>), required: >= oVirt <strong>4.2</strong></p>
```

Comment:
SSH keys can not be managed with recent oVirt <strong>{apiVersion}</strong> version. Please upgrade oVirt to <strong>4.2</strong> or higher.